module/VuFind/src/VuFind/Content/AbstractCover.php

```diff
@@ -145,4 +159,41 @@ public function supports($ids)
      * @return string|bool
      */
     abstract public function getUrl($key, $size, $ids);
+
+    /**
+     * Get cover metadata for a particular API key and set of IDs (or empty array).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object, 'issn' pointing to a string and 'oclc' pointing
+     * to an OCLC number string)
+     *
+     * @return array Array with keys: url, backlink_url, backlink_text
+     */
+    public function getMetadata(?string $key, string $size, array $ids)
+    {
+        $url = $this->getUrl($key, $size, $ids);
+        return $url ? ['url' => $url] : [];
+    }
+
+    /**
+     * Is mandatory to show backlink to cover source?
+     *
+     * @return bool
+     */
+    public function isBacklinkMandatory(): bool
+    {
+        return $this->mandatoryBacklinks;
+    }
+
+    /**
```

Comment:
Does it make sense to set this at the cover provider level? I was thinking it might make more sense as a global setting in config.ini. At least, it seems more likely to me that a user would want to turn off all cover backlinks in search results, than that one provider would want them there and another would not. Of course, there's also the question of how mandatory backlinks figure in; if backlinks are mandatory, should we always display them everywhere regardless of user preference?

Reply:
To be honest, it is designed for ObalkyKnih.cz - it is mandatory on biblio detail page according to theirs conditions of use.

But at least displaying the cover source backlink on other pages could be configured by user....